src/objects/zcl_abapgit_object_aifc.clas.abap

```diff
@@ -0,0 +1,509 @@
+        ifver2        TYPE c LENGTH 4,
+        ifver         TYPE c LENGTH 5,
+      END OF ty_icd_data_key.
+    TYPES:
+      BEGIN OF ty_table_data_s,
+        tabname    TYPE tabname,
+        table_data TYPE REF TO data,
+      END OF ty_table_data_s.
+    TYPES:
+      ty_table_data_t TYPE SORTED TABLE OF
+           ty_table_data_s WITH UNIQUE KEY tabname.
+
+    DATA ms_icd_data_key TYPE ty_icd_data_key_s.
+
+    METHODS handle_table_data
+      IMPORTING
+        !iv_tabname TYPE tabname
+        !it_data    TYPE STANDARD TABLE
+      RAISING
+        zcx_abapgit_exception.
+    METHODS authorization_check
+      IMPORTING
+        !io_log           TYPE REF TO zif_abapgit_log
+      RETURNING
+        VALUE(rv_success) TYPE abap_bool
+      RAISING
+        zcx_abapgit_exception.
+    METHODS get_content_compress
+      IMPORTING
+        !io_log     TYPE REF TO zif_abapgit_log
+        !is_ifkeys  TYPE ty_aif_key_s
+        !iv_package TYPE devclass
+      RAISING
+        zcx_abapgit_exception.
+    METHODS validate_interface
+      IMPORTING
+        !is_ifkeys        TYPE ty_aif_key_s
+      RETURNING
+        VALUE(rv_success) TYPE abap_bool
+      RAISING
+        zcx_abapgit_exception.
+    METHODS compress_interface
+      IMPORTING
+        !is_ifkeys        TYPE ty_aif_key_s
+        !io_log           TYPE REF TO zif_abapgit_log
+      RETURNING
+        VALUE(rv_success) TYPE abap_bool
+      RAISING
+        zcx_abapgit_exception.
+
+  PRIVATE SECTION.
+    TYPES:
+      BEGIN OF ty_content_s,
+        tabname TYPE tabname,
+      END OF ty_content_s.
+    TYPES:
+      ty_content_t TYPE STANDARD TABLE OF ty_content_s WITH NON-UNIQUE DEFAULT KEY.
+
+    DATA mo_abapgit_util TYPE REF TO object.
+ENDCLASS.
+
+
+
+CLASS ZCL_ABAPGIT_OBJECT_AIFC IMPLEMENTATION.
+
+
+  METHOD authorization_check.
+    DATA: lx_dyn_call_error TYPE REF TO cx_sy_dyn_call_error.
+    DATA: lx_root TYPE REF TO cx_root.
+
+    rv_success = abap_false.
+    TRY.
+        CALL METHOD mo_abapgit_util->('/AIF/IF_ABAPGIT_AIFC_UTIL~AUTHORIZATION_CHECK')
+          RECEIVING
+            rv_success = rv_success.
+
+      CATCH cx_sy_dyn_call_error INTO lx_dyn_call_error.
+        zcx_abapgit_exception=>raise( iv_text = TEXT-003
+                                      ix_previous = lx_dyn_call_error ).
+      CATCH cx_root INTO lx_root.
+        io_log->add_exception( ix_exc  = lx_root
+                               is_item = ms_item ).
+        RETURN.
+    ENDTRY.
+  ENDMETHOD.
+
+
+  METHOD compress_interface.
+    DATA: lx_dyn_call_error TYPE REF TO cx_sy_dyn_call_error.
+    DATA: lx_root TYPE REF TO cx_root.
+
+    TRY.
+        CALL METHOD mo_abapgit_util->('/AIF/IF_ABAPGIT_AIFC_UTIL~COMPRESS_INTERFACE')
+          EXPORTING
+            is_ifkeys  = is_ifkeys
+          RECEIVING
+            rv_success = rv_success.
+
+      CATCH cx_sy_dyn_call_error INTO lx_dyn_call_error.
+        zcx_abapgit_exception=>raise( iv_text = TEXT-003
```

Comment:
Text symbols are not supported by abapGit standalone version. Therefore we use hard coded texts instead.